engine/jobmaster/dm/openapi.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/pb"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/openapi"
+	dmpkg "github.com/pingcap/tiflow/engine/pkg/dm"
+)
+
+func (jm *JobMaster) initOpenAPI(router *gin.RouterGroup) {
+	wrapper := openapi.ServerInterfaceWrapper{
+		Handler: jm,
+	}
+	// copy from openapi.RegisterHandlersWithOptions
+	router.DELETE("/binlog/tasks/:task-name", wrapper.DMAPIDeleteBinlogOperator)
+
+	router.GET("/binlog/tasks/:task-name", wrapper.DMAPIGetBinlogOperator)
+
+	router.POST("/binlog/tasks/:task-name", wrapper.DMAPISetBinlogOperator)
+
+	router.GET("/config", wrapper.DMAPIGetJobConfig)
+
+	router.PUT("/config", wrapper.DMAPIUpdateJobConfig)
+
+	router.GET("/schema/tasks/:task-name", wrapper.DMAPIGetSchema)
+
+	router.PUT("/schema/tasks/:task-name", wrapper.DMAPISetSchema)
+
+	router.GET("/status", wrapper.DMAPIGetJobStatus)
+
+	router.PUT("/status", wrapper.DMAPIOperateJob)
+}
+
+// DMAPIGetJobStatus implements the api of get job status.
+func (jm *JobMaster) DMAPIGetJobStatus(c *gin.Context, params openapi.DMAPIGetJobStatusParams) {
+	var tasks []string
+	if params.Tasks != nil {
+		tasks = *params.Tasks
+	}
+	resp, err := jm.QueryJobStatus(c.Request.Context(), tasks)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, resp)
+}
+
+// DMAPIGetJobConfig implements the api of get job config.
+func (jm *JobMaster) DMAPIGetJobConfig(c *gin.Context) {
+	cfg, err := jm.GetJobCfg(c.Request.Context())
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	bs, err := cfg.Yaml()
```

Comment:
If the output is directly passed to Web frontend, I guess JSON is more common to be used. Will discuss later

Reply:
JSON is almost the only accepted format in our doc